addons/mail/models/ir_attachment.py

```diff
@@ -34,3 +36,17 @@ def register_as_main_attachment(self, force=True):
                     related_record.message_main_attachment_id = self
                 except AccessError:
                     pass
+
+    def _attachment_format(self, main_attachment=None, safari=None):
```

Comment:
I don't like the idea of having a keyword for a specific browser, it should be feature-based (e.g. imagine safari changes his behaviour starting another version or another browser has it), And it should be documented as well.

Reply:
I agree with you, but I m just moving code that already have this issue: 
https://github.com/odoo/odoo/blob/14.0/addons/mail/models/mail_message.py#L994

Reply:
@phenix-factory but before it was internals of a method that can easily evolve if the browser implementation changes with time. By making it a method parameter that other modules can call, you frooze it and make it official.